### PR TITLE
JUser docs

### PR DESCRIPTION
**Pull request by numerator**
### Description

-

### Related Jira Tickets

-

### Comments 

-

Please ensure all items are complete before requesting a review:

- [ ] **Code Completed**: All code for the feature or bugfix has been implemented.
- [ ] **Unit Tests Added**: Unit tests are written for all new functionality, covering edge cases where applicable.
- [ ] **Integration Tests (where necessary)**: Integration tests are added for interactions between modules or components.
- [ ] **All Tests Passing**: All unit and integration tests are passing.
- [ ] **Type Safety Checked**: Ensure TypeScript types are correctly defined and used. Avoid `any` types without proper justification.
- [ ] **Typedoc Comments Added**: Ensure all functions, classes, interfaces, and types are documented using TypeDoc syntax.
- [ ] **Code Formatting**: All code files are formatted according to the project’s standards (e.g., Prettier, ESLint) and no linting errors remain.
- [ ] **No Console Logs**: Remove any `console.log` or other debugging output from the code.
- [ ] **Documentation Updated**: Update relevant project documentation (e.g., README, configuration guides) if necessary.

---

### CODE REVIEW

File: docs/docs/concepts/users.md [hunk 1]
```diff
@@ -3,5 +3,94 @@ id: users
...
 
-All users (or "participants") in JustIn are represented by JUser objects. 
+All users (or "participants") in JustIn are represented by `JUser` objects. When adding new users to JustIn, you need to provide a `NewUserRecord` which is just a `JUser` without the JustIn-provided id.
+
+`JUser` and `NewUserRecord` type definitions:
+```ts
+export type JUser = {
+  id: string;
+  uniqueIdentifier: string;
+  attributes: Record<string, any>;
```
Comment (jadrubio) [inline docs/docs/concepts/users.md:15]: this makes it sound like there is a key named attributes that can hold a Record<string, any>

are we intending the top level a JUser to be

```
{
    id: 12123-1232-12312
    uniqueIdentifier: 'test-user'
    attributes: {
         ... all other things that make up a user
         }
  }
  ```

File: docs/docs/concepts/users.md [hunk 2]
```diff
@@ -3,5 +3,94 @@ id: users
...
+};
+
+export type NewUserRecord = {
+  uniqueIdentifier: string;
+  attributes: Record<string, any>;
```
Comment (jadrubio) [inline docs/docs/concepts/users.md:20]: same as above

File: docs/docs/concepts/users.md (hunk 1 above)
Comment (jadrubio) [reply docs/docs/concepts/users.md:15]: I see the example below confirms this

File: docs/docs/concepts/users.md (hunk 2 above)
Comment (jadrubio) [reply docs/docs/concepts/users.md:20]: same as above

File: docs/docs/concepts/users.md [hunk 3]
```diff
@@ -3,5 +3,94 @@ id: users
...
+
+The other essential aspect of each `JUser` object is the `attributes` field, which is a key-value map (aka `Record<string, any>`) that you can use to represent variables related to each user that are relevant to your study, including the operation of your DecisionRules and Tasks (e.g., contextual states used for taiolring, study arm labels used for activating different intervention components, etc.).
+
+The JustIn functions for working with users have the following signatures:
+```ts
+async addUsers (newUserRecords: NewUserRecord []): Promise<JUser []>;
+async addUser (newUserRecord: NewUserRecord): Promise<JUser>;
+async updateUser(attributeName: string, attributesToUpdate: Record<string, any>): Promise<JUser>;
```
[the remark this answers is not in the document]
Comment (peiyaoh) [reply docs/docs/concepts/users.md:32]: You are right. I paste the incorrect version to Mark.

```
async updateUser(uniqueIdentifier: string, attributesToUpdate: Record<string, any>): Promise<JUser>
```
